### PR TITLE
Add missing extendhfxf2 in compiler rt

### PR DESCRIPTION
**Pull request by biabbas**
Issue: #105181
extendhfxf2 calls extendhfXfy to convert _Float16 to double, then type casts this converted value to long double. 
__uint128_t may not be available on all architectures. Thus I din't use extendhfXfy to widen precision to 128 bits.

### CODE REVIEW

**Comment by biabbas** [comment]
Object file preview with the change in 34a812421f03c888cb7fbf3b7f27b4732c99afe8. 
```
0000000000000170 <__extendhfxf2>:
 170:   50                      push   %rax
 171:   66 0f c5 c0 00          pextrw $0x0,%xmm0,%eax
 176:   66 89 44 24 06          mov    %ax,0x6(%rsp)
 17b:   66 0f c4 44 24 06 00    pinsrw $0x0,0x6(%rsp),%xmm0
 182:   e8 99 fe ff ff          call   20 <__extendXfYf2__>
 187:   f3 0f 11 04 24          movss  %xmm0,(%rsp)
 18c:   d9 04 24                flds   (%rsp)
 18f:   58                      pop    %rax
 190:   c3                      ret    
 191:   66 66 66 66 66 66 2e    data16 data16 data16 data16 data16 cs nopw 0x0(%rax,%rax,1)
 198:   0f 1f 84 00 00 00 00 
 19f:   00 
```

File: compiler-rt/lib/builtins/extendhfsf2.c [hunk 1]
```diff
@@ -16,6 +16,10 @@ COMPILER_RT_ABI NOINLINE float __extendhfsf2(src_t a) {
   return __extendXfYf2__(a);
 }
 
+COMPILER_RT_ABI NOINLINE long double __extendhfxf2(src_t a) {
```
Comment (lntue) [inline compiler-rt/lib/builtins/extendhfsf2.c:19]: I think this should be in a separate `extendhfxf2.c` file.  This file is for converting `half -> single`.

Comment (lntue) [reply compiler-rt/lib/builtins/extendhfsf2.c:19]: `__extendXfYf2__` is defined in `fp_extend_impl.inc`, not in here.  You can define `SRC_HALF`, `DST_SINGLE`, and include `fp_extend_impl.inc` in that separate file.

Comment (biabbas) [reply compiler-rt/lib/builtins/extendhfsf2.c:19]: Ok. Sure.

**Review by lntue — CHANGES_REQUESTED** [review]
Can you add some simple test cases to `llvm-project/compiler-rt/test/builtins/Unit/extendhfxf2_test.c`?

**Comment by lntue** [comment]
> ⚠️ C/C++ code formatter, clang-format found issues in your code. ⚠️
> 
> You can test this locally with the following command:
> View the diff from clang-format here.

Can you run clang-format and sync to head to fix buildkite errors?  Thanks

File: compiler-rt/lib/builtins/extendhfxf2.c [hunk 2]
```diff
@@ -0,0 +1,19 @@
+//===-- lib/extendhfxf2.c - half -> long double conversion -------------*- C
+//-*-===//
```
Comment (lntue) [inline compiler-rt/lib/builtins/extendhfxf2.c:2]: Nit: make this fit in one 80-char line.

File: compiler-rt/test/builtins/Unit/extendhfxf2_test.c [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+           a, x, expected);
+  }
+  return ret;
+}
+
+char assumption_1[sizeof(_Float16) * CHAR_BIT == 16] = {0};
+
+int main() {
+  // Small positive value
+  if (test_extendhfxf2(0.09997558593750000000f, 0.09997558593750000000f))
```
Comment (lntue) [inline compiler-rt/test/builtins/Unit/extendhfxf2_test.c:26]: nit: use suffix `L` for second argument.

File: compiler-rt/test/builtins/Unit/extendhfxf2_test.c [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    return 1;
+
+  // Small negative value
+  if (test_extendhfxf2(-0.09997558593750000000f, -0.09997558593750000000f))
```
Comment (lntue) [inline compiler-rt/test/builtins/Unit/extendhfxf2_test.c:30]: nit: use suffix `L` for second argument.

File: compiler-rt/test/builtins/Unit/extendhfxf2_test.c [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    return 1;
+
+  // Zero
+  if (test_extendhfxf2(0.0f, 0.0L))
+    return 1;
+
+  // Smallest positive non-zero value
+  if (test_extendhfxf2(0x1p-16f, 0x1p-16f))
```
Comment (lntue) [inline compiler-rt/test/builtins/Unit/extendhfxf2_test.c:38]: nit: use suffix `L` for second argument.

File: compiler-rt/test/builtins/Unit/extendhfxf2_test.c [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    return 1;
+
+  // Smallest negative non-zero value
+  if (test_extendhfxf2(-0x1p-16f, -0x1p-16f))
```
Comment (lntue) [inline compiler-rt/test/builtins/Unit/extendhfxf2_test.c:42]: nit: use suffix `L` for second argument.

**Review by lntue — APPROVED** [review]

**Comment by lntue** [comment]
@biabbas  Look like some extra guards are needed.  I'm going to revert the PR and wait for your fix.

**Pull request merged**